templates/github/.github/workflows/ci.yml.j2

```
@@ -38,15 +38,45 @@ jobs:
           .github/workflows/scripts/check_commit.sh
       {%- endif %}
   {%- endif %}
+
+  check-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      run_tests: {{ '${{ steps.check.outputs.run_tests }}' }}
+    steps:
+      {{ checkout(depth=0, path=plugin_name) | indent(6) }}
+
+      {{ setup_python("3.12") | indent(6) }}
+
+      {{ install_python_deps(["gitpython"]) | indent(6) }}
+
+      - name: Analyze changed files
+        shell: bash
+        id: check
+        run: |
+          set +e
+          BASE_REF={{ "${{ github.event.pull_request.base.sha }}" }}
+          echo "Checking against:"
+          git name-rev $BASE_REF
+          python3 .ci/scripts/skip_tests.py . $BASE_REF
+          exit_code=$?
+          if [ $exit_code -ne 0 ] && [ $exit_code -ne 1 ]; then
+            echo "Error: skip_tests.py returned unexpected exit code $exit_code"
+            exit $exit_code
+          fi
+          echo "run_tests=$exit_code" >> $GITHUB_OUTPUT
+
   {%- if is_pulpdocs_member %}
 
   docs:
     uses: "./.github/workflows/docs.yml"
   {%- endif %}
 
   lint:
+    needs: check-changes
+    if: needs.check-changes.outputs.run_tests == '1'
     {%- if pre_job_template %}
-    needs: {{ pre_job_template.name }}
+    needs: [check-changes, {{ pre_job_template.name }}]
```

```suggestion
    needs:
      - "check-changes"
      {%- if pre_job_template %}
      - "{{ pre_job_template.name }}"
      {%- endif %}
    if: needs.check-changes.outputs.run_tests == '1'
```
The endif below needs to be removed...